core/Services/Calendar/EventStore.vala

```diff
@@ -330,6 +343,22 @@ public class Calendar.EventStore : Object {
             week_start = 7;
         }
 
+        // Accessing the locale settings
+        if (locale_settings != null) {
+            int week_start_user_pref = locale_settings.get_int ("first-day") + 1;
+
+            if (week_start >= 1 && week_start <= 7) {
+                if (week_start_user_pref == week_start_posix) {
+                    week_start = week_start_posix + glib_offset;
+                } else {
+                    print ("User Preference for week start: %i\nSystem Preference for week start: %i\n", week_start_user_pref, week_start);
+                    week_start = week_start_user_pref + glib_offset;
+                }
```

Comment:
I am a little concerned about this. Neither of these work use the Sunday special case. Also, doesn't line 351 just repeat the same calculation as line 341? Finally, this doesn't account for the fact that `glib_offset` isn't necessarily always the same. It depends on the value of the posix `week_1stday` variable. We need to take this into account either here or in switchboard. (All the locales I've found in Ubuntu use the same value, but that's not guaranteed.)

Reply:
You could tend to this in a newer PR, fixing both this here and in the datetime indicator.

Reply:
I suppose so. I'm worried that it won't work as expected in this state though. Have you tried it set to Sunday? It looks to me like the Sunday bug should come into effect on Ubuntu systems. At least when the system locale is set to something like en_US.

If I'm wrong though, then I don't think this would ever impact an Ubuntu based system. I'd like to submit a PR later as you say, but it wouldn't be necessary to function.

Reply:
There's no apparent bugs if I set it to Sunday. :confused:

Reply:
Ok, seems like we're good then? I guess this works for me and I can come back around to the posix weirdness later when I have time.

Reply:
Alright. :)